File: ios/xue_hua_speaker_earpiece_toggle/Sources/xue_hua_speaker_earpiece_toggle/AudioRouteNames.swift
import Foundation

enum AudioRouteNames {
  static let speaker = "speaker"
  static let earpiece = "earpiece"
  static let external = "external"
  static let unknown = "unknown"

  static let switchableRoutes: Set<String> = [speaker, earpiece]
}

struct RouteApplyResult {
  let applied: String
  let available: Bool

  var dictionary: [String: Any] {
    ["applied": applied, "available": available]
  }
}

enum AudioRouteError: Error {
  case invalidRoute(String)

  var flutterError: (code: String, message: String) {
    switch self {
    case .invalidRoute(let route):
      return ("INVALID_ROUTE", "Unknown audio route: \(route)")
    }
  }
}

File: android/src/main/kotlin/com/kurban/xue_hua_speaker_earpiece_toggle/AudioRouteNames.kt
package com.kurban.xue_hua_speaker_earpiece_toggle

internal object AudioRouteNames {
    const val SPEAKER = "speaker"
    const val EARPIECE = "earpiece"
    const val EXTERNAL = "external"
    const val UNKNOWN = "unknown"

    val SWITCHABLE_ROUTES = setOf(SPEAKER, EARPIECE)
}

internal data class RouteApplyResult(
    val applied: String,
    val available: Boolean
) {
    fun toMap(): Map<String, Any> = mapOf(
        "applied" to applied,
        "available" to available
    )
}
